### ISSUE TITLE
Implicit gtest_filter fails when running 300 tests

### ISSUE THREAD

**Issue by SaturnIC** [matepek/vscode-catch2-test-adapter#423]
**Checklist**

- [x] The issue is about this extension and NOT about a fork.
- [x] Checked the [ALL the SUPPORT](https://github.com/matepek/vscode-catch2-test-adapter/blob/master/documents/support.md) document.
- [x] The latest version of the extension was used.
- [x] It is not related to remote-vscode or I checked the following [issue](https://github.com/matepek/vscode-catch2-test-adapter/issues/201)
- [x] Imagine yourself into my position and think how hard to debug the issue without insufficient information.
      I understand that you have privacy concerns and I expect you to understand that this extension is developed for free.
      Thanks.

**Describe the bug**

- Running googletests on windows pc.
- All tests pass from the commandline (Just running the compiled exe)
- 9 / 300 Tests fail by running the test suite with test adapter out of vscode.
  - Found out the test invocation from test adapter uses `--gtest_filter=<test-suite>.<test-case>:...` parameters
    for every single test case(!)
    Running the test suite on the cli with the same filter parameters causes tests to fail on the cli as well
  - Found out the test invocation from test adapter uses `--gtest_color=no` parameter,
    disabling this parameter on the CLI lets only 1/300 fail
- Running the tests with `--gtest_filter=* --gtest_color=no` lets all tests pass
  => The gtest_filter parameter with many positive filter strings seems to cause flaky tests
- Running only the tests in the failed test suite by themselves (14 tests) in the VSCode test adapter GUI lets all those tests pass (they failed before)
- Question: Could the filter string in the "run all tests view" be disabled or replaced with * since it seems buggy to use gtest_filter with lots of tests?

**Comment by matepek** [comment]
Hello,

It can be that your tests are depending on each other and thats why it's flaky and not because of this extension.
You can try command line with [shuffle](https://google.github.io/googletest/advanced.html#shuffling-the-tests) to reproduce the same issue. You can try running the same command line with filter the extension does and figuring out why your tests are flaky.

This filtering feature is a base of other features which would not work so I need solid proof it is not working properly to start to work on some alternatives.

Let me know if you found the issue or need more advice, meanwhile I don't consider this an issue of the extension.

**Comment by SaturnIC** [comment]
Yes the tests are dependent, but the vscode filter M U S T not change the order of running those tests!

**Comment by matepek** [comment]
[Point 2](https://medium.com/@kasperjuunge/ten-principles-of-unit-testing-db2d8e0b203e)

**Comment by SaturnIC** [comment]
Only beginners care about some arbitrary rules. 
(Clean Code being a major example for the harm caused by philosophy pushed with religious fervor)

Software is all about predictability and deterministic behavior.

And vscode gtest extension shuffling execution orders is an issue that must be fixed.

**Comment by matepek** [comment]
GoogleTest framework does not define execution order (if I'm correct), therefore developers should not depend on it.

**Comment by SaturnIC** [comment]
The order is fixed after compilation of the tests,
so run them in the order they have been set in the binary.

**Comment by rokas-kbit** [comment]
We are also impacted by this. I commend extension maintainer to sticking to best practices. Unfortunately, in real life setting things are less ideal. Large enterprise software has so much inertia that at times it is impossible to fix anything. By all means please provide default settings that lead to development of best possible code, however when it is impossible to override these behaviors it does make extension simply unusable in certain environments that can not be fixed easily. Therefore i urge maintainer to reconsider allowing users to override the default behavior.

**Comment by matepek** [comment]
I understand this casing trouble but there are some features which are depend on it.

For example UI won’t know which tests will be executed so it can just update the currently running test cases.
Another example is parallel running. This extension can run test cases parallel but it only make sense if they are not depend on each other which they shouldn’t since GoogleTest framework does not specify execution order. 

Anyway,
I’m gonna think about some workaround for cases like these when code has undefined expectations/requirements.
